src/test/e2e/api/admin/environment.test.ts

```diff
@@ -101,3 +109,112 @@ test('Updating a non existing environment yields 404', async () => {
         .send({ displayName: 'Update this' })
         .expect(404);
 });
+
+test('Can update sort order', async () => {
+    const envName = 'sort-order-env';
+
+    await app.request
+        .post('/api/admin/environments')
+        .send({
+            name: envName,
+            displayName: 'Enable feature for environment',
+            type: 'production',
+        })
+        .set('Content-Type', 'application/json')
+        .expect(201);
+
+    await app.request
+        .put('/api/admin/environments/sort-order')
+        .send({
+            ':global:': 2,
+            [envName]: 1,
+        })
+        .expect(200);
+
+    await app.request
+        .get('/api/admin/environments')
+        .expect(200)
+        .expect('Content-Type', /json/)
+        .expect((res) => {
+            expect(res.body.environments[0].name).toBe(envName);
+            expect(res.body.environments[0].sortOrder).toBe(1);
+
+            expect(res.body.environments[1].name).toBe(':global:');
+            expect(res.body.environments[1].sortOrder).toBe(2);
+        });
+});
+
+test('Sort order will fail on wrong data format', async () => {
+    const envName = 'sort-order-env';
+
+    await app.request
+        .put('/api/admin/environments/sort-order')
+        .send({
+            ':global:': 'test',
+            [envName]: 1,
+        })
+        .expect(400);
+});
+
+test('Can update environment enabled status', async () => {
+    const envName = 'update-env';
+
+    await app.request
+        .post(`/api/admin/environments/${envName}/on`)
+        .set('Content-Type', 'application/json')
+        .expect(204);
+});
+
+test('Can update environment disabled status', async () => {
+    const envName = 'update-env';
+
+    await app.request
+        .post(`/api/admin/environments/${envName}/off`)
+        .set('Content-Type', 'application/json')
+        .expect(204);
+});
```

Comment:
Doesn't this test then assume that the `update-env` already exists due to having been created in an earlier test?
I can't see a setup method that always adds it. Yes, Jest currently runs tests within a file sequentially, but we've run into issues with this before. I think it would be better if we ensure that each test creates the environment it runs against, or one of our beforeAll/beforeEach methods creates a fixed set of environments that we then know exist through all the tests

Reply:
True. I will fix.